system_metrics/system_metrics_client_test.go

```diff
@@ -271,9 +210,12 @@ func TestCreateSystemMetricsRecord(t *testing.T) {
 			Compression: 6,
 			Schema:      SchemaTypeRawEvents,
 		})
-		require.Error(t, err)
+		assert.Error(t, err)
 		assert.Equal(t, id, "")
+		mc.mu.Lock()
```

Comment:
If we have to lock/unlock every time we get data anyways, can we just make a method to do a thread-safe read?